stl/inc/format

```diff
@@ -3554,8 +3562,18 @@ struct formatter {
 _FMT_P2286_BEGIN
 template <class _Ty, class _CharT, _Basic_format_arg_type _ArgType>
 struct _Formatter_base {
+private:
     using _Pc = basic_format_parse_context<_CharT>;
 
+public:
+#if _HAS_CXX23
+    constexpr void set_debug_format() noexcept
+        requires (_Is_debug_enabled_fmt_type(_ArgType))

```

Comment:
I'm slightly concerned that this affects program-defined types that inherit from `std::formatter<int>`. Although this function is constrained, its name can be found by name lookup.

```c++
void set_debug_format(auto&); // #1

struct my_formatter : std::formatter<int> {
    auto parse(auto& x) {
        set_debug_format(*this); // before: OK, calls #1
                                 // after: invalid. Name lookup finds the constrained member
        return x.begin();
    }
};
```